Fix incorrect mipmap size calculation for uncompressed DDS textures

Monstrofil opened the pull request:
Previously, the DDS texture loader incorrectly calculated mipmap dimensions for uncompressed textures when the original dimensions were non-power-of-two.

For example, given a 120×120 texture, the expected mipmap chain should be:
    mipmap=0, size=120×120
    mipmap=1, size=60×60
    mipmap=2, size=30×30
    mipmap=3, size=15×15
    mipmap=4, size=7×7
    mipmap=5, size=3×3
    mipmap=6, size=1×1

But the loader was producing:
    mipmap=0, size=120×120
    mipmap=1, size=60×60
    mipmap=2, size=30×30
    mipmap=3, size=15×15
    mipmap=4, size=8×8
    mipmap=5, size=4×4
    mipmap=6, size=2×2

This commit corrects the logic to properly compute width and height independently at each mip level.

The change corresponds to [how the official Microsoft dds loader works](https://github.com/microsoft/DirectXTex/blob/main/DDSTextureLoader/DDSTextureLoader11.cpp#L1082)  / (MIT license).


Here are two textures for testing.
[dds_mipmap_test.zip](https://github.com/user-attachments/files/19668019/dds_mipmap_test.zip)


Fixes #105136. I also added a bit deeper explanation [in comments of the issue](https://github.com/godotengine/godot/issues/105136#issuecomment-2789786371).

fire commented:
I think this looks good but I maybe @BlueCube3310 can verify

akien-mga commented:
For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

Monstrofil committed:
Fix incorrect mipmap size calculation for uncompressed DDS textures

Previously, the DDS texture loader incorrectly calculated mipmap dimensions
for uncompressed textures when the original dimensions
were non-power-of-two.

For example, given a 120×120 texture, the expected mipmap chain should be:
    mipmap=0, size=120×120
    mipmap=1, size=60×60
    mipmap=2, size=30×30
    mipmap=3, size=15×15
    mipmap=4, size=7×7
    mipmap=5, size=3×3
    mipmap=6, size=1×1

But the loader was producing:
    mipmap=0, size=120×120
    mipmap=1, size=60×60
    mipmap=2, size=30×30
    mipmap=3, size=15×15
    mipmap=4, size=8×8
    mipmap=5, size=4×4
    mipmap=6, size=2×2

This commit corrects the logic to properly compute width and height independently at each mip level.

Fixes issue #105136.

Monstrofil commented:
@akien-mga thanks, I force-pushed updated commit with correct email set.

Repiteo commented:
Thanks! Congratulations on your first merged contribution! 🎉
